### ISSUE TITLE
Local property sources not decrypted when binding to @ConfigurationProperties

### ISSUE THREAD

**Issue by shakuzen** [spring-cloud/spring-cloud-commons#269]
Using the Spring Cloud Config server, I want to be able to decrypt encrypted properties for the config server's own configuration properties. This works with `Dalston.SR4` and `Edgware.RC1` on Spring Boot 1.5.8, but it does not work with current Finchley snapshots and Spring Boot 2.0.0.M5. Note that remote property sources still have their properties decrypted properly.

Reproduction project: https://github.com/shakuzen/demo-config-server-encrypt

**Comment by dsyer** [comment]
Here's a test (passes with Spring Cloud context 1.x, fails with 2.x: 

```
	@Test
	public void symmetricPropertyValues() {
		ConfigurableApplicationContext context = new SpringApplicationBuilder(
				TestConfiguration.class).web(WebApplicationType.NONE).properties(
						"encrypt.key:pie",
						"foo.password:{cipher}bf29452295df354e6153c5b31b03ef23c70e55fba24299aa85c63438f1c43c95")
						.run();
		assertEquals("test", context.getEnvironment().getProperty("foo.password"));
	}

	@Test
	public void symmetricConfigurationProperties() {
		ConfigurableApplicationContext context = new SpringApplicationBuilder(
				TestConfiguration.class).web(WebApplicationType.NONE).properties(
						"encrypt.key:pie",
						"foo.password:{cipher}bf29452295df354e6153c5b31b03ef23c70e55fba24299aa85c63438f1c43c95")
						.run();
		assertEquals("test", context.getBean(PasswordProperties.class).getPassword());
	}

	@Configuration
	@EnableConfigurationProperties(PasswordProperties.class)
	protected static class TestConfiguration {

	}

	@ConfigurationProperties("foo")
	protected static class PasswordProperties {
		private String password;

		public String getPassword() {
			return password;
		}

		public void setPassword(String password) {
			this.password = password;
		}
	}
```

**Comment by dsyer** [comment]
Should be fixed here: https://github.com/spring-projects/spring-boot/commit/85b1511085bc395ce4ffa0d20287372f9e3100c3

**Comment by spencergibb** [comment]
We are now on boot snapshots, so the fix mentioned above should be available.

**Comment by shakuzen** [comment]
Yes, indeed, the repro project does not have the issue anymore with current snapshots. I believe this issue can be closed, but I think it would be a good idea to add the tests Dave made above so it is covered for the future.

**Comment by philwebb** [comment]
I've pushed a fix for https://github.com/spring-projects/spring-boot/issues/10840, let me know if it helps.

**Comment by consult-kk** [comment]
Hi,
One of the files added as part of the fix "CompositePropertySources.java"  doesnt seem to be available in SpringBoot 2.1.5.RELEASE . If this should not be used , then what is the alternative.
thanks

**Comment by dsyer** [comment]
The fix was in Spring Boot, so if they refactored to use something else we probably don't care. We don't have any direct need for `CompositePropertySources` here as far as I know.
